### PR TITLE
fix: Content restriction not working

### LINKED ISSUE #1225: Word count min limit not working
The min word count is made 600. But i see that less than 600 word count can be sent for "pending review". 
please use uid: ijdk7777  and pass: asdfg@123456789012
site https://www.innertowords.com/wp-login.php

### PR DESCRIPTION
**Pull request by sapayth**
Word/Character count limit not working
fix #1225

### CODE REVIEW

File: includes/class-frontend-form-post.php [hunk 1]
```diff
@@ -309,6 +309,42 @@ public function submit_post() {
...
+            $restriction_to = ! empty( $single_field['restriction_to'] ) ? $single_field['restriction_to'] : 'min';
+            $restriction_type = ! empty( $single_field['restriction_type'] ) ? $single_field['restriction_type'] : 'word';
+
+            $current_data = ! empty( $_POST[ $single_field['name'] ] ) ? sanitize_text_field( wp_unslash( $_POST[ $single_field['name'] ] ) ) : '';
+            $label = ! empty( $single_field['label'] ) ? $single_field['label'] : '';
+
+            // if restriction by character count
+            if ( 'character' === $restriction_type && 'min' === $restriction_to ) {
+                if ( strlen( $current_data ) > 0 && strlen( $current_data ) < $restricted_num ) {
+                    $this->send_error( 'Minimum ' . $restricted_num . ' character required for ' . $label );
```
Comment (tareq1988) [inline includes/class-frontend-form-post.php:328]: The error texts need to be translated.

Comment (sapayth) [reply includes/class-frontend-form-post.php]: > The error texts need to be translated.

Done

File: includes/class-frontend-form-post.php [hunk 2]
```diff
@@ -309,6 +309,42 @@ public function submit_post() {
...
+            $restriction_to = ! empty( $single_field['restriction_to'] ) ? $single_field['restriction_to'] : 'min';
+            $restriction_type = ! empty( $single_field['restriction_type'] ) ? $single_field['restriction_type'] : 'word';
+
+            $current_data = ! empty( $_POST[ $single_field['name'] ] ) ? sanitize_text_field( wp_unslash( $_POST[ $single_field['name'] ] ) ) : '';
+            $label = ! empty( $single_field['label'] ) ? $single_field['label'] : '';
+
+            // if restriction by character count
+            if ( 'character' === $restriction_type && 'min' === $restriction_to ) {
+                if ( strlen( $current_data ) > 0 && strlen( $current_data ) < $restricted_num ) {
+                    $this->send_error( __( 'Minimum ', 'wp-user-frontend' ) . $restricted_num . __( ' character required for ', 'wp-user-frontend' ) . $label );
```
Comment (tareq1988) [inline includes/class-frontend-form-post.php:328]: That's not how you translate a string with a variable. That's how you do it:

```php
sprintf( __( 'Minimum %d character is required for %s', 'wp-user-frontend' ), $restricted_num, $label );
```

File: assets/js/frontend-form.js [hunk 3]
```diff
@@ -949,10 +1018,10 @@
                     if ( limit && numWords > limit && 'max' === limit_to ) {
                         WP_User_Frontend.editorLimit.blockTyping(event);
                         WP_User_Frontend.markError( field, 'limit' );
-                        jQuery('.mce-path-item.mce-last', ed.container).html( wpuf_frontend['word_'+limit_to ] + numWords +'/'+ limit );
+                        jQuery('.mce-path-item.mce-last', ed.container).html( wpuf_frontend['word_'+limit_to ] + ' ' + numWords +'/'+ limit );
                     }else if ( limit && numWords < limit && 'min' === limit_to ) {
```
Comment (tareq1988) [inline assets/js/frontend-form.js:1022]: Fix the space

File: assets/js/frontend-form.js [hunk 4]
```diff
@@ -925,6 +937,63 @@
...
+                var errorMessage = wpuf_frontend['word_'+limit_to ] + ' ' + limit;
+
+                if ( (numWords > 1) && (numWords > limit) && ('max' === limit_to) ) {
+                    WP_User_Frontend.markError( field, 'limit' );
+
+                    if (isTinymce) {
+                        jQuery('.mce-path-item.mce-last', ed.container).html( wpuf_frontend['word_'+limit_to ] + ' ' + numWords +'/'+ limit );
+                    }
+                    return true;
+                }else if ( (numWords < limit) && ('min' === limit_to) && numWords > 1 ) {
```
Comment (tareq1988) [inline assets/js/frontend-form.js:982]: fix the space

**Review by tareq1988 — CHANGES_REQUESTED** [review]

File: includes/class-frontend-form-post.php [hunk 5]
```diff
@@ -309,6 +309,42 @@ public function submit_post() {
...
+            } elseif ( 'character' === $restriction_type && 'max' === $restriction_to ) {
+                if ( strlen( $current_data ) > 0 && strlen( $current_data ) > $restricted_num ) {
+                    $this->send_error( sprintf( __( 'Maximum %d character is allowed for %s', 'wp-user-frontend' ), $restricted_num, $label ) );
+                }
+            }
+
+            // if restriction by word count
+            if ( 'word' === $restriction_type && 'min' === $restriction_to ) {
+                if ( str_word_count( $current_data ) > 0 && str_word_count( $current_data ) < $restricted_num ) {
+                    $this->send_error( __( 'Minimum ', 'wp-user-frontend' ) . $restricted_num . __( ' word required for ', 'wp-user-frontend' ) . $label );
```
Comment (tareq1988) [inline includes/class-frontend-form-post.php:339]: You missed the translation improvement here and below.

**Pull request merged**